modules/core/04-channel/v2/keeper/keeper.go

```diff
@@ -252,6 +253,30 @@ func (k *Keeper) aliasV1Channel(ctx context.Context, portID, channelID string) (
 	return channelv2, true
 }
 
+// resolveV2Identifiers returns the client identifier and the counterpartyInfo for the client given the packetId
+// Note: For fresh eureka channels, the client identifier and packet identifier are the same.
+// For aliased channels, the packet identifier will be the original channel ID and the counterpartyInfo will be constructed from the channel
+func (k *Keeper) resolveV2Identifiers(ctx context.Context, portId string, packetId string) (string, clienttypes.CounterpartyInfo, error) {
```

Comment:
I think maybe the name could be better here. If I have to read the comment to understand what the function does for me, I consider it a code smell :)

Reply:
After seeing it used further down I am even more confused. Is it taking in packetId or a client id?

Reply:
Yea this was because it could also be an original channelID for aliasing, in which case I still need to retrieve the original clientiD